components/match2/wikis/teamfortress/match_group_input_custom.lua

```diff
@@ -0,0 +1,190 @@
+		and MatchFunctions.calculateMatchScore(games, match.bestof)
+		or nil
+	Array.forEach(opponents, function(opponent, opponentIndex)
+		opponent.score, opponent.status = MatchGroupInput.computeOpponentScore({
+			walkover = match.walkover,
+			winner = match.winner,
+			opponentIndex = opponentIndex,
+			score = opponent.score,
+		}, autoScoreFunction)
+	end)
+
+	match.finished = MatchGroupInput.matchIsFinished(match, opponents)
+
+	if match.finished then
+		match.resulttype = MatchGroupInput.getResultType(winnerInput, finishedInput, opponents)
+		match.walkover = MatchGroupInput.getWalkover(match.resulttype, opponents)
+		match.winner = MatchGroupInput.getWinner(match.resulttype, winnerInput, opponents)
+		MatchGroupInput.setPlacement(opponents, match.winner, 1, 2)
+	end
+
+	MatchFunctions.getTournamentVars(match)
+
+	match.stream = Streams.processStreams(match)
+	match.links = MatchFunctions.getLinks(match, games)
+
+	match.extradata = MatchFunctions.getExtraData(match)
+
+	match.games = games
+	match.opponents = opponents
+
+	return match
+end
+
+---@param match table
+---@param opponentCount integer
+---@return table[]
+function CustomMatchGroupInput.extractMaps(match, opponentCount)
+	local maps = {}
+	for key, map in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local finishedInput = map.finished --[[@as string?]]
+		local winnerInput = map.winner --[[@as string?]]
+
+		map.extradata = MapFunctions.getExtraData(map)
+		map.finished = MatchGroupInput.mapIsFinished(map)
+
+		local opponentInfo = Array.map(Array.range(1, opponentCount), function(opponentIndex)
+			local score, status = MatchGroupInput.computeOpponentScore({
+				walkover = map.walkover,
+				winner = map.winner,
+				opponentIndex = opponentIndex,
+				score = map['score' .. opponentIndex],
+			})
+			return {score = score, status = status}
+		end)
+
+		map.scores = Array.map(opponentInfo, Operator.property('score'))
+		if map.finished then
+			map.resulttype = MatchGroupInput.getResultType(winnerInput, finishedInput, opponentInfo)
+			map.walkover = MatchGroupInput.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInput.getWinner(map.resulttype, winnerInput, opponentInfo)
+		end
+
+		table.insert(maps, map)
+		match[key] = nil
+	end
+
+	return maps
+end
+
+CustomMatchGroupInput.processMap = FnUtil.identity
+
+--
+-- match related functions
+--
+
+---@param maps table[]
+---@param bestOf integer
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps, bestOf)
+	return function(opponentIndex)
+		return MatchGroupInput.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
+end
+
+---@param match table
+---@return integer
+function MatchFunctions.getBestOf(match)
+	local bestof = tonumber(Logic.emptyOr(match.bestof, Variables.varDefault('bestof')))
+	Variables.varDefine('bestof', bestof)
+	return bestof or DEFAULT_BESTOF
+end
```

Comment:
Common's best of doen't work for TF2?

Reply:
It seems to be used in this place (removing the getbestof function returns error) assuming for `|bestof=X` to work
![image](https://github.com/user-attachments/assets/2be4ccb6-bae3-44f4-af96-fb926bc540e3)

but TF doesnt have weird bestof, its just the standard BOs out there